internal/microsoft/data_microsoft_security.go

```diff
@@ -0,0 +1,133 @@
+package microsoft
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+	"github.com/blackstork-io/fabric/plugin/plugindata"
+)
+
+func makeMicrosoftSecurityDataSource(loader MicrosoftSecurityClientLoadFn) *plugin.DataSource {
+	return &plugin.DataSource{
+		Doc:      "The `microsoft_security` data source queries Microsoft Security API.",
+		DataFunc: fetchMicrosoftSecurity(loader),
+		Config: &dataspec.RootSpec{
+			Attrs: []*dataspec.AttrSpec{
+				{
+					Doc:         "The Azure client ID",
+					Name:        "client_id",
+					Type:        cty.String,
+					Constraints: constraint.RequiredNonNull,
+				},
+				{
+					Doc:    "The Azure client secret. Required if `private_key_file` or `private_key` is not provided.",
+					Name:   "client_secret",
+					Type:   cty.String,
+					Secret: true,
+				},
+				{
+					Doc:         "The Azure tenant ID",
+					Name:        "tenant_id",
+					Type:        cty.String,
+					Constraints: constraint.RequiredNonNull,
+				},
+				{
+					Doc:  "The path to the private key file. Ignored if `private_key` or `client_secret` is provided.",
+					Name: "private_key_file",
+					Type: cty.String,
+				},
+				{
+					Doc:  "The private key contents. Ignored if `client_secret` is provided.",
+					Name: "private_key",
+					Type: cty.String,
+				},
+				{
+					Doc:  "The key passphrase. Ignored if `client_secret` is provided.",
+					Name: "key_passphrase",
+					Type: cty.String,
+				},
+			},
+		},
+		Args: &dataspec.RootSpec{
+			Attrs: []*dataspec.AttrSpec{
+				{
+					Name:        "endpoint",
+					Doc:         "API endpoint to query",
+					Type:        cty.String,
+					Constraints: constraint.RequiredNonNull,
+					ExampleVal:  cty.StringVal("/users"),
+				},
+				{
+					Name: "query_params",
+					Doc:  "HTTP query parameters",
+					Type: cty.Map(cty.String),
+				},
+				{
+					Name:         "size",
+					Doc:          "Number of objects to be returned",
+					Type:         cty.Number,
+					Constraints:  constraint.NonNull,
+					DefaultVal:   cty.NumberIntVal(50),
+					MinInclusive: cty.NumberIntVal(1),
```

Comment:
This might need max value. I see ms clients use `min(size, defaultPageSizeGraph)` which would result always using `defaultPageSizeGraph` when `size` attribute is over 200.